lib/funding_offer.js

```diff
@@ -23,13 +24,65 @@ const FIELDS = {
 const FIELD_KEYS = Object.keys(FIELDS)
 
 class FundingOffer extends Model {
-  constructor (data = {}) {
+  /**
+  * @param {Object|Array} data - either a map of order fields or a raw array
+  * @param {Object} handler - optional, rest or websocket object capable of submitting funding offers
+  */
+  constructor (data = {}, handler) {
     super(data, FIELDS, BOOL_FIELDS, FIELD_KEYS)
+    this._handler = handler
   }
 
   static unserialize (arr) {
     return super.unserialize(arr, FIELDS, BOOL_FIELDS, FIELD_KEYS)
   }
+
+  /**
+  * Creates an order map that can be passed to the `on` command.
+  *
+  * @return {Object} o
+  */
+  toNewOfferPacket () {
+    return {
+      type: this.type,
+      symbol: this.symbol,
+      amount: prepareAmount(+this.amount),
+      rate: prepareAmount(+this.rate),
+      period: this.period,
+      flags: this.flags
+    }
+  }
+
+  /**
+   * @param {handler} handler - optional ws or rest, defaults to internal ws
+   * @return {Promise} p
+   */
+  submit (handler = this._handler) {
+    if (!handler) return Promise.reject(new Error('no submit handler'))
```

Comment:
This is minor, but the error strings could be rewritten as `'no API interface provided'` or something similar. Likewise, `this._handler` would be better as `this._apiInterface` or so, to clarify its functionality.